webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java

```diff
@@ -613,69 +632,99 @@ private static void addFmcSolutionSheet(PdfWriter docWriter, Document doc, Scram
+        rulesList.add("- "+messages.getString("rule5"));
+        rulesList.add("- "+messages.getString("rule6"));
+        rulesList.add("- "+messages.getString("rule7"));
+        rulesList.add("- "+messages.getString("rule8"));
+        rulesList.add("- "+messages.getString("rule9"));
+        int maxMoves = WCA_MAX_MOVES_FMC;
+        rulesList.add("- "+String.format(messages.getString("rule10"), maxMoves));
+        rulesList.add("- "+messages.getString("rule11"));
 
-        com.itextpdf.text.List rules = new com.itextpdf.text.List(com.itextpdf.text.List.UNORDERED);
-        rules.add("Notate your solution by writing one move per bar.");
-        rules.add("To delete moves, clearly erase/blacken them.");
-        rules.add("Face moves F, B, R, L, U, and D are clockwise.");
-        rules.add("Rotations x, y, and z follow R, U, and F.");
-        rules.add("' inverts a move; 2 doubles a move. (e.g.: U', U2)");
-        rules.add("w makes a face move into two layers. (e.g.: Uw)");
-        rules.add("A [lowercase] move is a cube rotation. (e.g.: [u])");
-
-        ct.addElement(rules);
         int rulesTop = competitorInfoBottom + (withScramble ? 55 : 143);
-        ct.setSimpleColumn(left+padding, scrambleBorderTop, competitorInfoLeft-padding, rulesTop, 0, Element.ALIGN_LEFT);
-        ct.go();
-
-        rules = new com.itextpdf.text.List(com.itextpdf.text.List.UNORDERED);
-        rules.add("You have 1 hour to find a solution.");
-        rules.add("Your solution length will be counted in OBTM.");
-        int maxMoves = WCA_MAX_MOVES_FMC;
-        rules.add("Your solution must be at most " + maxMoves + " moves, including rotations.");
-        rules.add("Your solution must not be directly derived from any part of the scrambling algorithm.");
-        ct.addElement(rules);
-        MAGIC_NUMBER = 150; // kill me now
-        ct.setSimpleColumn(left+padding, scrambleBorderTop, rulesRight-padding, rulesTop-MAGIC_NUMBER, 0, Element.ALIGN_LEFT);
+        Rectangle rulesRectangle = new Rectangle(left+padding, scrambleBorderTop, competitorInfoLeft-padding, rulesTop);
+        
+        float potentialFontSize = 15f;
+        // resize rules vertically
+        while (true){
+            int numberOfLines = 0;
+            for (String item : rulesList){
+                float substringWidth = bf.getWidthPoint(item, potentialFontSize);
+                numberOfLines += Math.ceil(substringWidth/rulesRectangle.getWidth());
+            }
+            
+            float totalHeight = numberOfLines*potentialFontSize;
+            
+            // 0.65 used to fit text, found by brute Force.
+            // a better approach would be considering the actual rendered font height
+            if (totalHeight < (0.65*rulesRectangle.getHeight())){
+                break;
+            }
+            potentialFontSize -= 0.1f;
+        }
+        
+        String rules = "";
+        for (String item : rulesList){
+            rules += item+"\n";
```

Comment:
This feels like you could do a string join here instead of a loop

Reply:
Ok :)